include/openthread/link.h

```diff
@@ -1126,6 +1126,77 @@ uint8_t otLinkGetWakeupChannel(otInstance *aInstance);
  */
 otError otLinkSetWakeupChannel(otInstance *aInstance, uint8_t aChannel);
 
+/**
+ * Enables or disables listening for wake-up frames.
+ *
+ * Requires `OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE`.
+ *
+ * @param[in]  aInstance     A pointer to an OpenThread instance.
+ * @param[in]  aEnable       true to enable listening for wake-up frames, or false otherwise.
+ *
+ * @retval OT_ERROR_NONE          Successfully enabled / disabled the listening for wake-up frames.
+ * @retval OT_ERROR_INVALID_ARGS  The listen duration is greater than the listen interval.
+ * @retval OT_ERROR_INVALID_STATE Could not enable listening for wake-up frames due to bad configuration.
+ */
+otError otLinkSetWedListenEnabled(otInstance *aInstance, bool aEnable);
```

Comment:
Can we add a wake up callback here to customize the behavior after a device is waken up?
```suggestion
otError otLinkSetWakeUpListenEnabled(otInstance *aInstance, otWakeUpCallback aCallback);
```

Reply:
You mean like this, right?
`otError otLinkSetWakeUpListenEnabled(otInstance *aInstance, bool aEnable, otWakeUpCallback aCallback);`

But is there really a need to control the behavior from the application? According to the spec, once waken up the WED should start MLE attach to the WC. Only if succeeded there will be an `OT_CHANGED_THREAD_ROLE` signal sent to the application, which then can check whether it's connected to WC or not using some future MLE API.

I would say that the callback configuration can be added later on if needed with something like:
`otError otLinkSetWakeUpListenCallback(otInstance *aInstance, otWakeUpCallback aCallback);`